carla/recourse_methods/autoencoder/models.py

```diff
@@ -0,0 +1,150 @@
+import os
+from typing import Callable, List, Optional
+
+import numpy as np
+from keras import backend as K
+from keras.layers import Dense, Input
+from keras.models import Model, Sequential
+
+
+def layers_valid(layers: List) -> bool:
+    """
+    Checks if the layers parameter has at least minimal requirements
+
+    Returns
+    -------
+    bool
+    """
+    if len(layers) < 2:
+        return False
+
+    for layer in layers:
+        if layer <= 0:
+            return False
+
+    return True
+
+
+class Autoencoder:
+    def __init__(
+        self,
+        layers: List,
+        data_name: str,
+        optimizer: str = "rmsprop",
+        loss: Optional[Callable] = None,
+    ) -> None:
+        """
+        Defines the structure of the autoencoder
+
+        Parameters
+        ----------
+        layers : list(int > 0)
+            Depending on the position and number elements, it determines the number and width of layers in the form of
+            [input_layer, hidden_layer_1, ...., hidden_layer_n, latent_dimension]
+        data_name : str
+            Name of the dataset. Is used for saving model.
+        loss: Callable, optional
+            Loss function for autoencoder model. Default is Binary Cross Entropy.
+        optimizer: str, optional
```

Comment:
Now that we have typehints, i'd say its a bit double (maintance) work to also have it in the comments, i'd be fine to leave it out here

Reply:
👍